src/components/application_manager/src/commands/hmi/response_from_hmi.cc

```diff
@@ -72,12 +73,8 @@ void ResponseFromHMI::SendResponseToMobile(
 void ResponseFromHMI::CreateHMIRequest(
     const hmi_apis::FunctionID::eType& function_id,
     const smart_objects::SmartObject& msg_params) const {
-  smart_objects::SmartObjectSPtr result = new smart_objects::SmartObject;
-
-  if (!result) {
-    SDL_ERROR("Memory allocation failed.");
-    return;
-  }
+  smart_objects::SmartObjectSPtr result =
+      utils::MakeShared<smart_objects::SmartObject>();
```

Comment:
@Kozoriz `utils::MakeShared` uses `std:: nothrow` form of `new`. It throws no `std::bad_alloc` and returns `NULL` instead. [operator new](http://en.cppreference.com/w/cpp/memory/new/operator_new) So the `SmartObjectSPtr` may be empty.